react/tic-tac-toe/src/components/Game/index.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+
+import Board from './components/Board';
+import styles from './styles.module.scss';
+
+class Game extends Component {
+  state = {
+    history: [{
+      squares: Array(9).fill(null)
+    }],
+    stepNumber: 0,
+    xIsNext: true
+  };
+  
+  handleClick = (i) => {
+    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = {...current.squares};
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([{
+        squares: squares,
+      }]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext
+    })
+  };
+
+  jumpTo(step){
+    this.setState({
+      stepNumber: step,
+      xIsNext: (step % 2) === 0,
+    });
+  }
+
+  render() {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const winner = calculateWinner(current.squares);
+
+    const moves = history.map((step, move) => {
+      const desc = move ? 'Go to move # ' + move : 'Go to game start';
+      return (
+        <li key={move}>
+          <button onClick={ () => this.jumpTo(move) }> {desc} </button>
+        </li>
+      )
+    });
+
+    let status;
+    if (winner) 
+      status = 'Winner: ' + winner;
+    else 
+      status = 'Next player: ' + (this.state.xIsNext ? 'X' : 'O');
+
+    return (
+      <div className={styles.game}>
+        <div className={styles.gameBoard}>
+          <Board 
+            squares={current.squares}
+            onClick={(i) => this.handleClick(i)}
+          />
+        </div>
+        <div className={styles.gameInfo}>
+          <div>{status}</div>
+          <ol>{moves}</ol>
+        </div>
+      </div>
+    );
+  }
+}
+
+function calculateWinner(squares) {
```

Comment:
You can move this function to a `gameUtils` file (or whatever name you want). This will help the code look better and will comply with the single responsibility principle. 😺

Reply:
Please keep in mind this comment. 👀